deepchem/data/data_loader.py

```diff
@@ -437,6 +439,124 @@ def _featurize_shard(self, shard):
     return (X, np.ones(len(X), dtype=bool))
 
 
+class JsonLoader(DataLoader):
+  """
+  Creates `Dataset` objects from input json files. 
+
+  This class provides conveniences to load data from json files.
+  It's possible to directly featurize data from json files using
+  pandas, but this class may prove useful if you're processing
+  large json files that you don't want to manipulate directly in
+  memory.
+
+  It is meant to load JSON files formatted as "records" in line
+  delimited format, which allows for sharding.
+  ``list like [{column -> value}, ... , {column -> value}]``.
+
+  Examples
+  --------
+  >> import pandas as pd
+  >> df = pd.DataFrame(some_data)
+  >> df.columns.tolist()
+  .. ['formula', structure', 'task']
+  >> df.to_json('file.json', orient='records', lines=True)
+  >> loader = JsonLoader(['task'], {'structure': dict}, 'formula')
+  >> dataset = loader.create_dataset('file.json')
+  
+  """
+
+  def __init__(self,
+               tasks: List[str],
+               json_fields: Dict[str, type],
+               id_field: str = None,
+               featurizer: Optional[Featurizer] = None,
+               log_every_n: int = 1000):
+    """Initializes JsonLoader.
+
+    Parameters
+    ----------
+    tasks : List[str]
+      List of task names
+    json_fields : Dict[str, type]
+      column names and dtypes in dataframe containing data to be featurized
+      e.g. {"structure": dict, "composition": str}
```

Comment:
I don't understand how this argument is supposed to work.  As far as I can tell it never even looks at the values in the dict, just the keys.

For each sample, a dataset contains four fields: features, labels, weights, and id.  So I was expecting something that would tell it which JSON field to use for each one.

Reply:
Ok got it - my apologies, I was a bit confused about the purpose of introducing the argument. If it is a mapping between those four fields and the JSON, it might be better to just have four general arguments to `JsonLoader`: `feature_field`, `label_field`, `weights_field`, and `id_field` that specify the JSON fields.

Reply:
I like the structure with the 4 general argument. 👍

Reply:
Great, I will implement that!